File: Amazon/ServerSide/Amazon.Solution/Amazon.Services/ProductService/Dto/ProductDto.cs
using Microsoft.AspNetCore.Http;
using System.ComponentModel.DataAnnotations;


namespace Amazon.Services.ProductService.Dto
{
	public class ProductDto
	{
	
		public string Name { get; set; }
		public string Description { get; set; }
		public decimal Price { get; set; }
		public IFormFile ImageFile { get; set; }
		public int QuantityInStock { get; set; }
		public int BrandId { get; set; }
		public int CategoryId { get; set; }
		public DiscountDto Discount { get; set; }
		public ICollection<IFormFile> ImagesFiles { get; set; }
    }
}

File: Amazon/ClientSide/Amazon.ClientSide/src/app/Models/productDto.ts
import { Discount } from "./Discount";

export class ProductDto {
    Name: string;
    Description: string;
    Price: number;
    ImageFile: File;
    QuantityInStock: number;
    BrandId: number;
    CategoryId: number;
    Discount:Discount;
    ImagesFiles: File[];

    constructor
        (
            Name: string,
            Price: number,
            Description: string,
            ImageFile: File,
            QuantityInStock: number,
            BrandId: number,
            CategoryId: number,
            Discount:Discount,
            ImagesFiles: File[],
        ) {
        this.Name = Name;
        this.Price = Price;
        this.Description = Description;
        this.ImageFile = ImageFile;
        this.QuantityInStock = QuantityInStock;
        this.BrandId = BrandId;
        this.CategoryId = CategoryId;
        this.Discount = Discount;
        this.ImagesFiles = ImagesFiles;
    }
}
